lib/aws/active_job/sqs/configuration.rb

```diff
@@ -17,20 +19,42 @@ class Configuration
           excluded_deduplication_keys: ['job_id']
         }.freeze
 
-        # @api private
-        attr_accessor :queues, :max_messages, :visibility_timeout,
-                      :shutdown_timeout, :client, :logger,
-                      :async_queue_error_handler, :message_group_id
+        GLOBAL_ENV_CONFIGS = %i[
+          threads backpressure
+          max_messages shutdown_timeout
+          visibility_timeout message_group_id
+        ].freeze
 
-        attr_reader :excluded_deduplication_keys
+        QUEUE_ENV_CONFIGS = %i[
+          url max_messages
+          visibility_timeout message_group_id
+        ].freeze
 
         # Don't use this method directly: Configuration is a singleton class, use
         # +Aws::ActiveJob::SQS.config+ to access the singleton config.
         #
+        # This class provides a Configuration object for AWS ActiveJob
+        # by pulling configuration options from Runtime, the ENV, a YAML file,
```

Comment:
I think ENV should take precedence over YAML, so that you can set ENV for the config file location. See how this was done in the sessionstore gem recently.

Reply:
ENV does take precedence over YAML and you can set ENV for the config file location. 

The order of precedence follows what we use in the SDK: code, ENV, file.

I notice that session store does NOT follow that pattern and instead uses ENV, code, file.  However, with usage in rails (and configuraton through code normally done in eg rails initializers, maybe ENV taking precedence over explicit code configuration is more useful).